libs/string_util/include/hpx/string_util/trim.hpp

```diff
@@ -0,0 +1,38 @@
+//  Copyright (c) 2020 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_STRING_UTIL_TRIM_HPP
+#define HPX_STRING_UTIL_TRIM_HPP
+
+#include <algorithm>
+#include <cstddef>
+#include <string>
+
+namespace hpx { namespace string_util {
+    template <typename CharT, class Traits, class Alloc>
+    void trim(std::basic_string<CharT, Traits, Alloc>& s)
+    {
+        auto first = std::find_if_not(std::cbegin(s), std::cend(s),
+            [](unsigned char c) { return std::isspace(c); });
```

Comment:
`unsigned char` does not look right here, did you mean `char`. Also, in other spots you use `int` as the argument for the lambda, should this be unified?

Reply:
Thanks! It should also be `int` (`isspace` takes an int).